timeseries/src/autogluon/timeseries/models/autogluon_tabular/transforms.py

```diff
@@ -13,29 +13,32 @@
     TIMESTAMP,
     TimeSeriesDataFrame,
 )
-from autogluon.timeseries.transforms.target_scaler import LocalTargetScaler, get_target_scaler
+from autogluon.timeseries.transforms.target_scaler import TargetScaler, get_target_scaler
 
 from .utils import MLF_ITEMID, MLF_TIMESTAMP
 
 
 class MLForecastScaler(BaseTargetTransform):
-    def __init__(self, scaler_type: Literal["standard", "robust", "min_max", "mean_abs"]):
+    def __init__(self, scaler_type: Literal["standard", "min_max", "mean_abs", "robust"]):
         # For backward compatibility
-        self.scaler_type = scaler_type
-        self.ag_scaler: Optional[LocalTargetScaler] = None
+        self.scaler_type: Literal["standard", "min_max", "mean_abs", "robust"] = scaler_type
+        self.ag_scaler: TargetScaler
 
     def _df_to_tsdf(self, df: pd.DataFrame) -> TimeSeriesDataFrame:
-        return df.rename(columns={self.id_col: ITEMID, self.time_col: TIMESTAMP}).set_index([ITEMID, TIMESTAMP])
+        return TimeSeriesDataFrame(
+            df.rename(columns={self.id_col: ITEMID, self.time_col: TIMESTAMP}).set_index([ITEMID, TIMESTAMP])
+        )
 
     def _tsdf_to_df(self, ts_df: TimeSeriesDataFrame) -> pd.DataFrame:
         return pd.DataFrame(ts_df).reset_index().rename(columns={ITEMID: self.id_col, TIMESTAMP: self.time_col})
 
-    def fit_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+    def fit_transform(self, df: pd.DataFrame) -> pd.DataFrame:  # type: ignore
         self.ag_scaler = get_target_scaler(name=self.scaler_type, target=self.target_col)
-        transformed = self.ag_scaler.fit_transform(self._df_to_tsdf(df)).reset_index()
```

Comment:
Was this a bug before?

Reply:
no, but it happens anyway within `_tsdf_to_df`